src/components/inputs/SearchSelect.jsx

```diff
@@ -0,0 +1,427 @@
+      break;
+    case KEY_CODES.ARROW_DOWN:
+      this._suggestNextOption(this.state.filterValue, this.state.suggestedOption);
+      break;
+    }
+  },
+
+  _onGroupClick(group, e) {
+    this._cancelBlurInterval();
+    this._toggleGroupOpenState(group);
+  },
+
+  _onOptionSelect(option, e) {
+    this._selectOption(option);
+  },
+
+  //Renders
+
+  render() {
+    let {
+      className,
+      placeHolder,
+      valueLink: {value},
+    } = this.props;
+    let { isFocused, filterValue } = this.state;
+    let filteredOptions = this._getFilteredOptions(filterValue);
+
+    return (
+      <div
+        className={classNames('Select', `Select--${isFocused ? 'opened' : 'closed'}`, className)}
+        onMouseLeave={this._onBlur}
+      >
+        <div className="Select-input"
+          onClick={this._onSelectValueClick}
+        >
+          <input
+            className={classNames('Select-filterInput', !filterValue && 'Select-filterInput-isEmpty')}
+            type="text"
+            ref="searchInput"
+            value={filterValue}
+            onFocus={this._onFocus}
+            onBlur={this._onFilterInputBlur}
+            onChange={this._onFilterInputChange}
+            onKeyDown={this._onFilterInputKeyDown}
+          />
+          <span className="Select-valueSpan">{value ? value.label : ''}</span>
+          {!filterValue && !value &&
+            <span className="Select-placeholder">{placeHolder}</span>
+          }
+        </div>
+        <div
+          className="Select-optionsList"
+          onMouseEnter={this._onFocus}
+        >
+          {_.map(filteredOptions, (option, i) => {
+            let render = option.isGroup ? this._renderGroup : this._renderOption;
+            return render(option, i);
+          })}
+        </div>
+      </div>
+    );
+  },
+
+  _renderGroup(group, key) {
+    let isOpened = this._isGroupOpen(this.state.openGroupsId, group);
+    return (
+      <div
+        className={classNames("Select-group", `Select-group--${isOpened ? 'opened' : 'closed'}`)}
+        key={key}
+      >
+        <div
+          className="Select-groupHeader"
+          onClick={this._onGroupClick.bind(null, group)}
+        >
+          <i className="Select-groupHeaderIcon" />
+          <span className="Select-groupheaderLabel">{group.label}</span>
+        </div>
+        <div className="Select-groupOptions">
+          {_.map(group.options, this._renderOption)}
+        </div>
+      </div>
+    );
+  },
+
+  _renderOption(option, key) {
+    let {
+      optionRender: OptionRender,
+    } = this.props;
+    let { filterValue, suggestedOption } = this.state;
+
+    let isSuggested = suggestedOption && suggestedOption.id === option.id;
+
+    return (
+      <OptionRender
+        className={classNames('Select-option', isSuggested && 'Select-option--suggested')}
+        key={key}
+        option={option}
+        filter={filterValue}
+        onClick={this._onOptionSelect.bind(null, option)}
```

Comment:
That's called atomic functions.
It's much more easier to read because you can just read the functions name and you now what it does. It gets better in later commits.
Plus, even if `_onOptionSelect` only call `selectOption` I know in the class methods that it's a listener whereas `selectOption` could be called from anywhere.